File: src/main/java/pl/training/workshop/m6/s14_adapter/step2/PaymentGateway.java
package pl.training.workshop.m6.s14_adapter.step2;

import pl.training.workshop.m6.s14_adapter.PaymentResult;
import pl.training.workshop.shared.Money;

/** Krok 2: preferowany interfejs kina - kwota jako Money, odmowa jako wynik, nie wyjątek. */
public interface PaymentGateway {
    PaymentResult pay(String reservationId, Money amount);
}

File: typescript/src/workshop/m6/s14_adapter/step3/PaymentGateway.ts
import type { Money } from '../../../shared/Money.js';
import type { PaymentResult } from '../PaymentResult.js';

/** Krok 3: preferowany interfejs kina - kwota jako Money, odmowa jako wynik, nie wyjątek. */
export interface PaymentGateway {
  pay(reservationId: string, amount: Money): PaymentResult;
}
